src/components/Actions.tsx

```diff
@@ -264,6 +269,9 @@ export const ShapesSwitcher = ({
               onImageAction({ pointerType });
             }
           }}
+          onContextMenu={(event, source) => {
+            onContextMenu && onContextMenu(event, source);
+          }}
```

Comment:
trying the custom contextMenu to the toolbar is also arbitrary coupling. Right we don't even support contextMenu other than on canvas, but it's something we may consider as part of https://github.com/excalidraw/excalidraw/issues/6139.

That said, for your specific case of switching between toolbar tool types, it should be implemented by a tool dropdown, e.g. similar to desktop editors like Affinity:

![affinity-toolbar-dropdown](https://user-images.githubusercontent.com/5153846/213866134-5dddaeb5-c8aa-4356-ae34-94109b1872aa.gif)

We'll need to make a decision on that yet: https://github.com/excalidraw/excalidraw/issues/6140

Reply:
I agree about using tool dropdowns.  My specific case does not involve switching between tool types.  Rather, I permit configuring seldom-modified properties of a single tool.  E.g. I use LaTeX on some boards and AsciiMath on other boards.  But I don't use both on the same board.

Reply:
I have removed this.